mvebu: Add support for wd my cloud ex2 ultra

nicefile opened the pull request:
same hardware as wd my cloud mirror gen2 but 1GB RAM

need testers since I don't have hw

edit:

Hardware
--------
Marvell Armada 385 (MV88F6820)
1GB RAM
256MB NAND
1x 1Gbit
2x USB 3.0
2x SATA-III
UART: 115200 8N1 3.3V
RTC
Weltrend MCU WT6703F connected via UART1 for Power LED / PWM Fan / hw reset / WoL

Installation
------------
Connect UART 3.3V adapter to JP2  pins: 1-RX / 2-GND / 5-TX

Use USB2.0 FAT32 pendrive with openwrt-mvebu-cortexa9-wd_cloud-ex2-ultra-initramfs-kernel.bin
 1. stop boot by pressing 1
 2. usb start
 3. fatload usb 0:1  0x02000000 openwrt-mvebu-cortexa9-wd_cloud-ex2-ultra-initramfs-kernel.bin;bootm 0x02000000 -
 4. do backup mtd1 mtd3
 5. use sysupgrade

Or tftp
 1. stop boot by pressing 1
 2. setenv ethact egiga2;setenv serverip 192.168.11.114;setenv ipaddr 192.168.11.113
 3. tftpboot 0x02000000 openwrt-mvebu-cortexa9-wd_cloud-ex2-ultra-initramfs-kernel.bin; bootm 0x02000000 -
 4. do backup mtd1 mtd3
 5. use sysupgrade

or Evgeny Kolesnikov <evgenyz@gmail.com> method from his failed PR 2040

- Using original firmware's network settings obtain SSH access to the device.
- Put *-image-cfs-factory.bin and *-uImage-factory.bin images into device's /tmp directory.
- Write kernel (uImage) image 'flash_eraseall /dev/mtd1 && nandwrite --markbad -p /dev/mtd1 /tmp/*-uImage-factory.bin'.
- Write rootfs (image-cfs) image 'ubiformat /dev/mtd3 -f /tmp/*-image.cfs-factory.bin -y'.
- Reboot the device.

Installation (upgrade):

Use *-sysupgrade.bin in a usual way.
Weltrend MCU control is done via uart1 19200
install coreutils-stty
stty -F /dev/ttyS1 raw speed 19200
stty -F /dev/ttyS1 raw speed 19200

PWM Fan Control
off: 00
echo -n -e '\xfa\x02\x00\x00\x00\x00\xfb' > /dev/ttyS1 slow: 5F
echo -n -e '\xfa\x02\x00\x5f\x00\x00\xfb' > /dev/ttyS1 max: FF
echo -n -e '\xfa\x02\x00\xff\x00\x00\xfb' > /dev/ttyS1

Power LED Control
Blue
echo -n -e '\xfa\x26\x00\x11\x00\x01\xfb' > /dev/ttyS1
Red
echo -n -e '\xfa\x26\x00\x14\x00\x01\xfb' > /dev/ttyS1
Orange
echo -n -e '\xfa\x26\x00\x12\x00\x01\xfb' > /dev/ttyS1

more here: https://github.com/c-MM/mcm-daemon/blob/master/mcm.h

evgenyz commented:
I see that you've picked the fallen flag.

evgenyz commented:
I wonder if it makes sense to also finish and integrate https://gitlab.com/evgenyz/wtmcu. Because on top of LED and Fan management it could potentially manage WOL and keep an eye on disks. Just like `mcm-daemon`.

nicefile commented:
> I wonder if it makes sense to also finish and integrate https://gitlab.com/evgenyz/wtmcu. Because on top of LED and Fan management it could potentially manage WOL and keep an eye on disks. Just like `mcm-daemon`.

nice to hear from you. For now we have basic support for mirror gen2 that I have and can test. Although I couldn't get WoL working .Same goes for power off via MCU (qnap-poweroff). In my opinion 'wtmcu' is not vital to the device support. Although working (out of the box) fan_control would be great.

evgenyz commented:
> For now we have basic support for mirror gen2 that I have and can test. Although I couldn't get WoL working .Same goes for power off via MCU (qnap-poweroff). 

Yeah, at that time I haven't managed to upstream DTS and driver, unfortunately.

> In my opinion 'wtmcu' is not vital to the device support. Although working (out of the box) fan_control would be great.

While it is not vital, it could: fix WOL (not implemented yet), control fan in a robust way, control LEDs in a meaningful and possibly advanced way (like signalling HDD problems) and probably something else, I don't quite remember now. But it definitely can reside in packages repo.

evgenyz commented:
BTW, it is not `Ext2 Ultra`, but `EX2 Ultra`.

nicefile commented:
@evgenyz name corrected. Btw  do you still have this NAS . Can you test this build  ?

evgenyz commented:
Yeah, I still have it somewhere. It might take a bit to set it up back, though. I might have a bit of spare time next weekend.

hauke commented:
I converted the PR to draft. Please mark it ready for review when you found a tester.

Commit:
mvebu: Add support for WD MyCloud EX2 Ultra

Hardware
--------
Marvell Armada 385 (MV88F6820)
1GB RAM
256MB NAND
1x 1Gbit
2x USB 3.0
2x SATA-III
UART: 115200 8N1 3.3V
RTC
Weltrend MCU WT6703F connected via UART1 for Power LED / PWM Fan / hw reset / WoL

Installation
------------
Connect UART 3.3V adapter to JP2  pins: 1-RX / 2-GND / 5-TX

Use USB2.0 FAT32 pendrive with openwrt-mvebu-cortexa9-wd_cloud-ex2-ultra-initramfs-kernel.bin
 1. stop boot by pressing 1
 2. usb start
 3. fatload usb 0:1  0x02000000 openwrt-mvebu-cortexa9-wd_cloud-ex2-ultra-initramfs-kernel.bin;bootm 0x02000000 -
 4. do backup mtd1 mtd3
 5. use sysupgrade

Or tftp
 1. stop boot by pressing 1
 2. setenv ethact egiga2;setenv serverip 192.168.11.114;setenv ipaddr 192.168.11.113
 3. tftpboot 0x02000000 openwrt-mvebu-cortexa9-wd_cloud-ex2-ultra-initramfs-kernel.bin; bootm 0x02000000 -
 4. do backup mtd1 mtd3
 5. use sysupgrade

or Evgeny Kolesnikov <evgenyz@gmail.com> method from his failed PR 2040

- Using original firmware's network settings obtain SSH access to the device.
- Put *-image-cfs-factory.bin and *-uImage-factory.bin images into device's /tmp directory.
- Write kernel (uImage) image 'flash_eraseall /dev/mtd1 && nandwrite --markbad -p /dev/mtd1 /tmp/*-uImage-factory.bin'.
- Write rootfs (image-cfs) image 'ubiformat /dev/mtd3 -f /tmp/*-image.cfs-factory.bin -y'.
- Reboot the device.

Installation (upgrade):

Use *-sysupgrade.bin in a usual way.
Weltrend MCU control is done via uart1 19200
install coreutils-stty
stty -F /dev/ttyS1 raw speed 19200
stty -F /dev/ttyS1 raw speed 19200

PWM Fan Control
off: 00
echo -n -e '\xfa\x02\x00\x00\x00\x00\xfb' > /dev/ttyS1
slow: 5F
echo -n -e '\xfa\x02\x00\x5f\x00\x00\xfb' > /dev/ttyS1
max: FF
echo -n -e '\xfa\x02\x00\xff\x00\x00\xfb' > /dev/ttyS1

Power LED Control
Blue
echo -n -e '\xfa\x26\x00\x11\x00\x01\xfb' > /dev/ttyS1
Red
echo -n -e '\xfa\x26\x00\x14\x00\x01\xfb' > /dev/ttyS1
Orange
echo -n -e '\xfa\x26\x00\x12\x00\x01\xfb' > /dev/ttyS1

more here: https://github.com/c-MM/mcm-daemon/blob/master/mcm.h

Signed-off-by: Robert Senderek <robert.senderek@10g.pl>
Link: https://github.com/openwrt/openwrt/pull/17779
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

Benetsu commented:
I'm the tester. My EX2 Ultra is currently running a build of openwrt provided by nicefile.
Here's dmesg and some other logs:
```
root@OpenWrt:~# dmesg
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 6.12.62 (user@DESKTOP-C15O2A1) (arm-openwrt-linux-muslgnueabi-gcc (OpenWrt GCC 14.3.0 r32143-4976b39b75) 14.3.0, GNU ld (GNU Binutils) 2.44) #0 SMP Fri Dec 26 08:10:37 2025
[    0.000000] CPU: ARMv7 Processor [414fc091] revision 1 (ARMv7), cr=10c5387d
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: WD MyCloud Ex2 Ultra
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x000000002fffffff]
[    0.000000]   HighMem  [mem 0x0000000030000000-0x000000003fffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000003fffffff]
[    0.000000] OF: reserved mem: Reserved memory: No reserved-memory node in the DT
[    0.000000] percpu: Embedded 14 pages/cpu s24716 r8192 d24436 u57344
[    0.000000] pcpu-alloc: s24716 r8192 d24436 u57344 alloc=14*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 
[    0.000000] Kernel command line:  nullparameter=
[    0.000000] Bootloader command line (ignored): root=/dev/ram console=ttyS0,115200
[    0.000000] Unknown kernel command line parameters "nullparameter=", will be passed to user space.
[    0.000000] Dentry cache hash table entries: 131072 (order: 7, 524288 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 262144
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=4 to nr_cpu_ids=2.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=2
[    0.000000] RCU Tasks Trace: Setting shift to 1 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=2.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] L2C: DT/platform modifies aux control register: 0x06070000 -> 0x16070000
[    0.000000] L2C-310 enabling early BRESP for Cortex-A9
[    0.000000] L2C-310 full line of zeros enabled for Cortex-A9
[    0.000000] L2C-310 D prefetch enabled, offset 1 lines
[    0.000000] L2C-310 dynamic clock gating enabled, standby mode enabled
[    0.000000] L2C-310 Coherent cache controller enabled, 16 ways, 1024 kB
[    0.000000] L2C-310 Coherent: CACHE_ID 0x410054c9, AUX_CTRL 0x56070001
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000001] sched_clock: 64 bits at 666MHz, resolution 1ns, wraps every 4398046511103ns
[    0.000016] clocksource: arm_global_timer: mask: 0xffffffffffffffff max_cycles: 0x999999d70a, max_idle_ns: 440795225152 ns
[    0.000032] Switching to timer-based delay loop, resolution 1ns
[    0.000200] Ignoring duplicate/late registration of read_current_timer delay
[    0.000204] clocksource: armada_370_xp_clocksource: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 76450417870 ns
[    0.000381] Calibrating delay loop (skipped), value calculated using timer frequency.. 1332.00 BogoMIPS (lpj=6660000)
[    0.000389] CPU: Testing write buffer coherency: ok
[    0.000413] CPU0: Spectre v2: using BPIALL workaround
[    0.000417] pid_max: default: 32768 minimum: 301
[    0.005731] Mount-cache hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.005740] Mountpoint-cache hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.008063] CPU0: thread -1, cpu 0, socket 0, mpidr 80000000
[    0.008697] Setting up static identity map for 0x100000 - 0x100060
[    0.008787] mvebu-soc-id: MVEBU SoC ID=0x6820, Rev=0x4
[    0.008873] mvebu-pmsu: Initializing Power Management Service Unit
[    0.008964] rcu: Hierarchical SRCU implementation.
[    0.008967] rcu: 	Max phase no-delay instances is 1000.
[    0.009117] Timer migration: 1 hierarchy levels; 8 children per group; 1 crossnode level
[    0.009281] smp: Bringing up secondary CPUs ...
[    0.009567] Booting CPU 1
[    0.009715] CPU1: thread -1, cpu 1, socket 0, mpidr 80000001
[    0.009723] CPU1: Spectre v2: using BPIALL workaround
[    0.009809] smp: Brought up 1 node, 2 CPUs
[    0.009814] SMP: Total of 2 processors activated (2664.00 BogoMIPS).
[    0.009819] CPU: All CPU(s) started in SVC mode.
[    0.010032] Memory: 1006704K/1048576K available (9216K kernel code, 657K rwdata, 2208K rodata, 16384K init, 243K bss, 39800K reserved, 0K cma-reserved, 262144K highmem)
[    0.013160] VFP support v0.3: implementor 41 architecture 3 part 30 variant 9 rev 4
[    0.013227] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.013237] futex hash table entries: 512 (order: 3, 32768 bytes, linear)
[    0.015950] pinctrl core: initialized pinctrl subsystem
[    0.017379] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.017941] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.018500] thermal_sys: Registered thermal governor 'step_wise'
[    0.018537] cpuidle: using governor menu
[    0.018676] mvebu-pmsu: CPU hotplug support is currently broken on Armada 38x: disabling
[    0.018682] mvebu-pmsu: CPU idle is currently broken on Armada 38x: disabling
[    0.028476] cryptd: max_cpu_qlen set to 1000
[    0.030178] SCSI subsystem initialized
[    0.030295] libata version 3.00 loaded.
[    0.030403] usbcore: registered new interface driver usbfs
[    0.030439] usbcore: registered new interface driver hub
[    0.030459] usbcore: registered new device driver usb
[    0.031289] clocksource: Switched to clocksource arm_global_timer
[    0.035824] NET: Registered PF_INET protocol family
[    0.035958] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.036939] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.036955] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.036963] TCP established hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.037008] TCP bind hash table entries: 8192 (order: 5, 131072 bytes, linear)
[    0.037146] TCP: Hash tables configured (established 8192 bind 8192)
[    0.037427] MPTCP token hash table entries: 1024 (order: 2, 16384 bytes, linear)
[    0.037595] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.037624] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    0.037943] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.037966] PCI: CLS 0 bytes, default 64
[    0.039926] workingset: timestamp_bits=14 max_order=18 bucket_order=4
[    0.049473] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.049482] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.053078] bounce: pool size: 64 pages
[    0.055001] armada-38x-pinctrl f1018000.pinctrl: registered pinctrl driver
[    0.055800] gpio gpiochip0: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.055937] debugfs: Directory 'f1018100.gpio' with parent 'regmap' already present!
[    0.056265] gpio gpiochip1: Static allocation of GPIO base is deprecated, use dynamic allocation.
[    0.056383] debugfs: Directory 'f1018140.gpio' with parent 'regmap' already present!
[    0.131231] mv_xor f1060800.xor: Marvell shared XOR driver
[    0.171968] mv_xor f1060800.xor: Marvell XOR (Descriptor Mode): ( xor cpy intr )
[    0.172175] mv_xor f1060900.xor: Marvell shared XOR driver
[    0.221901] mv_xor f1060900.xor: Marvell XOR (Descriptor Mode): ( xor cpy intr )
[    0.222168] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.222686] printk: legacy console [ttyS0] disabled
[    0.242921] f1012000.serial: ttyS0 at MMIO 0xf1012000 (irq = 38, base_baud = 12500000) is a 16550A
[    0.242949] printk: legacy console [ttyS0] enabled
[    0.981630] f1012100.serial: ttyS1 at MMIO 0xf1012100 (irq = 39, base_baud = 12500000) is a 16550A
[    0.993998] loop: module loaded
[    0.997967] ahci-mvebu f10a8000.sata: supply ahci not found, using dummy regulator
[    1.005701] ahci-mvebu f10a8000.sata: supply phy not found, using dummy regulator
[    1.014536] nand: device found, Manufacturer ID: 0xad, Chip ID: 0xda
[    1.020933] nand: Hynix H27U2G8F2CTR-BC
[    1.024804] nand: 256 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    1.032699] Bad block table found at page 131008, version 0x01
[    1.038879] Bad block table found at page 130944, version 0x01
[    1.044920] nand_read_bbt: bad block at 0x000009bc0000
[    1.054068] 8 fixed-partitions partitions found on MTD device pxa3xx_nand-0
[    1.061081] Creating 8 MTD partitions on "pxa3xx_nand-0":
[    1.066526] 0x000000000000-0x000000500000 : "U-Boot"
[    1.071785] 0x000000500000-0x000000a00000 : "kernel"
[    1.076962] 0x000000a00000-0x000000f00000 : "uRamdisk"
[    1.082389] 0x000000f00000-0x00000c800000 : "ubi"
[    1.087648] 0x00000c800000-0x00000d700000 : "rescue fw"
[    1.093260] 0x00000d700000-0x00000eb00000 : "config"
[    1.098640] 0x00000eb00000-0x00000f500000 : "reserve1"
[    1.104220] 0x00000f500000-0x00000ff00000 : "reserve2"
[    1.117418] hwmon hwmon0: temp1_input not attached to any thermal zone
[    1.125102] mvneta_bm f10c8000.bm: Buffer Manager for network controller enabled
[    1.134480] mvneta f1034000.ethernet eth0: Using random mac address 36:3c:05:f0:22:3a
[    1.142814] orion-ehci f1058000.usb: EHCI Host Controller
[    1.148257] orion-ehci f1058000.usb: new USB bus registered, assigned bus number 1
[    1.155946] orion-ehci f1058000.usb: irq 42, io mem 0xf1058000
[    1.181394] orion-ehci f1058000.usb: USB 2.0 started, EHCI 1.00
[    1.187772] hub 1-0:1.0: USB hub found
[    1.191579] hub 1-0:1.0: 1 port detected
[    1.196184] usbcore: registered new interface driver usb-storage
[    1.202819] armada38x-rtc f10a3800.rtc: registered as rtc0
[    1.208628] i2c_dev: i2c /dev entries driver
[    1.213090] mv64xxx_i2c f1011000.i2c: can't get pinctrl, bus recovery not supported
[    1.221033] mv64xxx_i2c f1011100.i2c: can't get pinctrl, bus recovery not supported
[    1.230849] orion_wdt: Initial timeout 171 sec
[    1.235674] sdhci: Secure Digital Host Controller Interface driver
[    1.241911] sdhci: Copyright(c) Pierre Ossman
[    1.246437] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.252776] marvell-cesa f1090000.crypto: CESA device successfully registered
[    1.260918] NET: Registered PF_INET6 protocol family
[    1.266841] Segment Routing with IPv6
[    1.270562] In-situ OAM (IOAM) with IPv6
[    1.274582] NET: Registered PF_PACKET protocol family
[    1.279674] 8021q: 802.1Q VLAN Support v1.8
[    1.283927] Registering SWP/SWPB emulation handler
[    1.302183] ahci-mvebu f10a8000.sata: supply ahci not found, using dummy regulator
[    1.309908] ahci-mvebu f10a8000.sata: supply phy not found, using dummy regulator
[    1.320767] usb_phy_generic usb3_0_phy: dummy supplies not allowed for exclusive requests (id=vbus)
[    1.330094] usb_phy_generic usb3_1_phy: dummy supplies not allowed for exclusive requests (id=vbus)
[    1.339392] ahci-mvebu f10a8000.sata: supply ahci not found, using dummy regulator
[    1.347107] ahci-mvebu f10a8000.sata: supply phy not found, using dummy regulator
[    1.355083] ahci-mvebu f10a8000.sata: masking port_map 0x3 -> 0x3
[    1.361246] ahci-mvebu f10a8000.sata: AHCI vers 0001.0000, 32 command slots, 6 Gbps, platform mode
[    1.370288] ahci-mvebu f10a8000.sata: 2/2 ports implemented (port mask 0x3)
[    1.377304] ahci-mvebu f10a8000.sata: flags: 64bit ncq sntf led only pmp fbs pio slum part sxs 
[    1.387731] scsi host0: ahci-mvebu
[    1.391759] scsi host1: ahci-mvebu
[    1.395341] ata1: SATA max UDMA/133 mmio [mem 0xf10a8000-0xf10a9fff] port 0x100 irq 52 lpm-pol 0
[    1.404205] ata2: SATA max UDMA/133 mmio [mem 0xf10a8000-0xf10a9fff] port 0x180 irq 52 lpm-pol 0
[    1.413343] xhci-hcd f10f0000.usb3: xHCI Host Controller
[    1.418698] xhci-hcd f10f0000.usb3: new USB bus registered, assigned bus number 2
[    1.426487] xhci-hcd f10f0000.usb3: hcc params 0x0a000990 hci version 0x100 quirks 0x0000000000000010
[    1.435853] xhci-hcd f10f0000.usb3: irq 43, io mem 0xf10f0000
[    1.441747] xhci-hcd f10f0000.usb3: xHCI Host Controller
[    1.447106] xhci-hcd f10f0000.usb3: new USB bus registered, assigned bus number 3
[    1.454691] xhci-hcd f10f0000.usb3: Host supports USB 3.0 SuperSpeed
[    1.461492] hub 2-0:1.0: USB hub found
[    1.465324] hub 2-0:1.0: 1 port detected
[    1.469541] usb usb3: We don't know the algorithms for LPM for this host, disabling LPM.
[    1.477979] hub 3-0:1.0: USB hub found
[    1.481787] hub 3-0:1.0: 1 port detected
[    1.486050] xhci-hcd f10f8000.usb3: xHCI Host Controller
[    1.491444] xhci-hcd f10f8000.usb3: new USB bus registered, assigned bus number 4
[    1.499062] xhci-hcd f10f8000.usb3: hcc params 0x0a000990 hci version 0x100 quirks 0x0000000000000010
[    1.508384] xhci-hcd f10f8000.usb3: irq 44, io mem 0xf10f8000
[    1.514246] xhci-hcd f10f8000.usb3: xHCI Host Controller
[    1.519593] xhci-hcd f10f8000.usb3: new USB bus registered, assigned bus number 5
[    1.527165] xhci-hcd f10f8000.usb3: Host supports USB 3.0 SuperSpeed
[    1.533885] hub 4-0:1.0: USB hub found
[    1.537712] hub 4-0:1.0: 1 port detected
[    1.541946] usb usb5: We don't know the algorithms for LPM for this host, disabling LPM.
[    1.550373] hub 5-0:1.0: USB hub found
[    1.554176] hub 5-0:1.0: 1 port detected
[    1.558578] UBI: auto-attach mtd3
[    1.561926] ubi0: attaching mtd3
[    1.742607] ata2: SATA link down (SStatus 0 SControl 300)
[    1.748065] ata1: SATA link down (SStatus 0 SControl 300)
[    1.761364] usb 2-1: new high-speed USB device number 2 using xhci-hcd
[    2.050592] ubi0: scanning is finished
[    2.111733] ubi0: attached mtd3 (name "ubi", size 185 MiB)
[    2.117257] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    2.124183] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    2.131010] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    2.138015] ubi0: good PEBs: 1479, bad PEBs: 1, corrupted PEBs: 0
[    2.144150] ubi0: user volume: 2, internal volumes: 1, max. volumes count: 128
[    2.151418] ubi0: max/mean erase counter: 3/1, WL threshold: 4096, image sequence number: 1233605412
[    2.160601] ubi0: available PEBs: 0, total reserved PEBs: 1479, PEBs reserved for bad PEB handling: 39
[    2.169972] ubi0: background thread "ubi_bgt0d" started, PID 626
[    2.170520] block ubiblock0_0: created from ubi0:0(rootfs)
[    2.181549] ubiblock: device ubiblock0_0 (rootfs) set to be root filesystem
[    2.188653] clk: Disabling unused clocks
[    2.203069] Freeing unused kernel image (initmem) memory: 16384K
[    2.209321] Run /init as init process
[    2.213014]   with arguments:
[    2.213018]     /init
[    2.213022]   with environment:
[    2.213025]     HOME=/
[    2.213028]     TERM=linux
[    2.213031]     nullparameter=
[    2.402635] init: Console is alive
[    2.406151] init: - watchdog -
[    2.413302] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    2.420669] gpio_button_hotplug: loading out-of-tree module taints kernel.
[    2.428674] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    2.439847] init: - preinit -
[    5.971300] random: crng init done
[    6.921312] usb 2-1: device descriptor read/64, error -110
[    7.183074] usb-storage 2-1:1.0: USB Mass Storage device detected
[    7.189516] scsi host2: usb-storage 2-1:1.0
[    7.321374] mvneta f1034000.ethernet eth0: PHY [f1072004.mdio-mii:00] driver [Marvell 88E1510] (irq=POLL)
[    7.331145] mvneta f1034000.ethernet eth0: configuring for phy/sgmii link mode
[    8.231585] scsi 2:0:0:0: Direct-Access     General  UDisk            5.00 PQ: 0 ANSI: 2
[    8.291003] sd 2:0:0:0: [sda] 245760 512-byte logical blocks: (126 MB/120 MiB)
[    8.298407] sd 2:0:0:0: [sda] Write Protect is off
[    8.303239] sd 2:0:0:0: [sda] Mode Sense: 0b 00 00 08
[    8.303364] sd 2:0:0:0: [sda] No Caching mode page found
[    8.308708] sd 2:0:0:0: [sda] Assuming drive cache: write through
[    8.371957]  sda: sda1
[    8.374444] sd 2:0:0:0: [sda] Attached SCSI removable disk
[   11.601945] mvneta f1034000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[   11.614579] mvneta f1034000.ethernet eth0: Link is Down
[   11.626913] procd: - early -
[   11.629876] procd: - watchdog -
[   12.152087] procd: - watchdog -
[   12.155384] procd: - ubus -
[   12.209604] procd: - init -
[   12.373881] kmodloader: loading kernel modules from /etc/modules.d/*
[   12.586135] kmodloader: done loading kernel modules from /etc/modules.d/*
[   12.758371] urngd: v1.0.2 started.
[   23.331458] mvneta f1034000.ethernet eth0: PHY [f1072004.mdio-mii:00] driver [Marvell 88E1510] (irq=POLL)
[   23.341247] mvneta f1034000.ethernet eth0: configuring for phy/sgmii link mode
[   23.483317] br-lan: port 1(eth0) entered blocking state
[   23.488593] br-lan: port 1(eth0) entered disabled state
[   23.493894] mvneta f1034000.ethernet eth0: entered allmulticast mode
[   23.500420] mvneta f1034000.ethernet eth0: entered promiscuous mode
[   27.671934] mvneta f1034000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[   27.680024] br-lan: port 1(eth0) entered blocking state
[   27.685296] br-lan: port 1(eth0) entered forwarding state
[   32.471303] v5.0-sata0: disabling
[   32.474652] v12.0-sata0: disabling
[   32.478072] v5.0-sata1: disabling
[   32.481425] v12.0-sata1: disabling
root@OpenWrt:~# cat /proc/meminfo
MemTotal:        1025160 kB
MemFree:          972468 kB
MemAvailable:     937644 kB
Buffers:               0 kB
Cached:            19172 kB
SwapCached:            0 kB
Active:            21480 kB
Inactive:              0 kB
Active(anon):      21480 kB
Inactive(anon):        0 kB
Active(file):          0 kB
Inactive(file):        0 kB
Unevictable:           0 kB
Mlocked:               0 kB
HighTotal:        262144 kB
HighFree:         235716 kB
LowTotal:         763016 kB
LowFree:          736752 kB
SwapTotal:             0 kB
SwapFree:              0 kB
Dirty:                 0 kB
Writeback:             0 kB
AnonPages:          2312 kB
Mapped:             3668 kB
Shmem:             19172 kB
KReclaimable:       2016 kB
Slab:              10480 kB
SReclaimable:       2016 kB
SUnreclaim:         8464 kB
KernelStack:         872 kB
PageTables:          332 kB
SecPageTables:         0 kB
NFS_Unstable:          0 kB
Bounce:                0 kB
WritebackTmp:          0 kB
CommitLimit:      512580 kB
Committed_AS:      27132 kB
VmallocTotal:     245760 kB
VmallocUsed:        3280 kB
VmallocChunk:          0 kB
Percpu:              160 kB
root@OpenWrt:~# ubus call system board
{
	"kernel": "6.12.62",
	"hostname": "OpenWrt",
	"system": "ARMv7 Processor rev 1 (v7l)",
	"model": "WD MyCloud Ex2 Ultra",
	"board_name": "wd,cloud-ex2-ultra",
	"rootfs_type": "initramfs",
	"release": {
		"distribution": "OpenWrt",
		"version": "SNAPSHOT",
		"firmware_url": "https://downloads.openwrt.org/",
		"revision": "r32341+68-d519a3ee86",
		"target": "mvebu/cortexa9",
		"description": "OpenWrt SNAPSHOT r32341+68-d519a3ee86",
		"builddate": "1766736637"
	}
}
```

nicefile commented:
@hauke this PR is ready for review.
